val.py

```diff
@@ -283,16 +294,25 @@ def run(
         # Metrics
         for si, pred in enumerate(preds):
             labels = targets[targets[:, 0] == si, 1:]
-            if tagged_data:
-                tagged_labels = targets[:, -1]
-                gt_boxes = targets[:, 2:-1] / torch.tensor((width, height, width, height), device=device)
             nl, npr = labels.shape[0], pred.shape[0]  # number of labels, predictions
             path, shape = Path(paths[si]), shapes[si][0]
+            try:
+                # TODO do we want to do this everytime in the for loop??
```

Comment:
We can also do "path = Path(paths[0])" before this for loop. Then we create these variables once per dataloader iteration